sdk/storage/Azure.Storage.DataMovement/src/Shared/JobPlanExtensions.cs

```diff
@@ -284,5 +242,83 @@ internal static string DictionaryToString(this IDictionary<string, string> dict)
             }
             return value;
         }
+
+        internal static JobPlanStatus ToJobPlanStatus(this DataTransferStatus transferStatus)
+        {
+            if (transferStatus == default)
+            {
+                return JobPlanStatus.None;
+            }
+
+            JobPlanStatus jobPlanStatus = (JobPlanStatus)Enum.Parse(typeof(JobPlanStatus), transferStatus.State.ToString());
+            if (transferStatus.HasFailedItems)
+            {
+                jobPlanStatus |= JobPlanStatus.HasFailed;
+            }
+            if (transferStatus.HasSkippedItems)
+            {
+                jobPlanStatus |= JobPlanStatus.HasSkipped;
+            }
+
+            return jobPlanStatus;
+        }
+
+        internal static DataTransferStatus ToDataTransferStatus(this JobPlanStatus jobPlanStatus)
+        {
+            DataTransferState state;
+            if (jobPlanStatus.HasFlag(JobPlanStatus.Queued))
+            {
+                state = DataTransferState.Queued;
+            }
+            else if (jobPlanStatus.HasFlag(JobPlanStatus.InProgress))
+            {
+                state = DataTransferState.InProgress;
+            }
+            else if (jobPlanStatus.HasFlag(JobPlanStatus.Pausing))
+            {
+                state = DataTransferState.Pausing;
+            }
+            else if (jobPlanStatus.HasFlag(JobPlanStatus.Stopping))
+            {
+                state = DataTransferState.Stopping;
+            }
+            else if (jobPlanStatus.HasFlag(JobPlanStatus.Paused))
+            {
+                state = DataTransferState.Paused;
+            }
+            else if (jobPlanStatus.HasFlag(JobPlanStatus.Completed))
+            {
+                state = DataTransferState.Completed;
+            }
+            else
+            {
+                state = DataTransferState.None;
+            }
+
+            bool hasFailed = jobPlanStatus.HasFlag(JobPlanStatus.HasFailed);
+            bool hasSkipped = jobPlanStatus.HasFlag(JobPlanStatus.HasSkipped);
+
+            return new DataTransferStatusInternal(state, hasFailed, hasSkipped);
+        }
+
+        internal static void WriteVariableLengthFieldInfo(
+            BinaryWriter writer,
+            byte[] bytes,
+            ref int currentVariableLengthIndex)
```

Comment:
can we get a quick docstring that describes this will increment the referenced value?